### PR TITLE
fix: allow creating lifecycle static cohorts

### LINKED ISSUE #15554: Save as cohort on lifecycle persons doesn't work
## Bug description

If you create a lifecycle insight, click on the persons modal, and try to create a cohort out of it with the "save as cohort" button, it returns empty cohorts

It doesn't look like we support lifecycle persons modal cohort saves right now. We should either add that functionality or remove the button from the UI for it.

## Environment

- [ ] PostHog Cloud US, project ID: [please provide from https://app.posthog.com/project/settings#project-id]
- [ ] PostHog Cloud EU, project ID: [please provide from https://eu.posthog.com/project/settings#project-id]
- [ ] PostHog Hobby self-hosted with `docker compose`, version/commit: [please provide]
- [ ] PostHog self-hosted with Kubernetes (deprecated, see ["Sunsetting Kubernetes support"](https://posthog.com/blog/sunsetting-helm-support-posthog)), version/commit: [please provide]

### PR DESCRIPTION
**Pull request by psykomal**
## Problem

Fixes #15554 . Create static cohort from lifecycle insight using persons modal
This page mentions create cohort using this feature - https://posthog.com/docs/product-analytics/lifecycle

## Changes

Added LifecycleActors and LifecycleFilter classes. Used in api/cohort.py::insert_cohort_actors_into_ch

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

## How did you test this code?

Manually creating cohorts in lifecycle insight
Added tests

### CODE REVIEW

**Comment by psykomal** [comment]
@liyiy Could you take a look at this or add the right POC for a review. It fixes #15554

**Review by neilkakkar — CHANGES_REQUESTED** [review]
Hey @psykomal - sorry it took a while for me to get to this!

This is mostly great! - Although there seems to be some duplication going on here. Have a look at `PathActors` and compare it to `LifecycleActors`. This place: https://github.com/PostHog/posthog/blob/master/posthog/api/person.py#L665 - shouldn't be separate from the new `LifecycleActor` class you're generating.

I'd recommend combining these into one, so there's only one code path to get persons for lifecycle, which is used by both cohorts & person querying.

File: posthog/queries/trends/lifecycle_actors.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        ).get_query()
+
+        lifecycle_type = self._filter.lifecycle_type
+        if lifecycle_type is None:
+            raise ValidationError("Must include lifecycle type")
+        target_date = self._filter.target_date
+        if target_date is None:
+            raise ValidationError("Must include specified target date")
+
+        return (
```
Comment (neilkakkar) [inline posthog/queries/trends/lifecycle_actors.py:29]: when you reconcile the two, you'll find that you can use `LIFECYCLE_QUERY_PEOPLE` sql from https://github.com/PostHog/posthog/blob/master/posthog/queries/trends/lifecycle.py#L61 here

Comment (neilkakkar) [reply posthog/queries/trends/lifecycle_actors.py:29]: instead of re-writing the SQL here

**Comment by psykomal** [comment]
@neilkakkar Have made the changes. Do have a look and let me know.

File: posthog/queries/trends/lifecycle.py [hunk 2]
```diff
@@ -31,6 +27,8 @@
 
 
 class Lifecycle:
+    # actor_query_class = LifecycleActors
```
Comment (neilkakkar) [inline posthog/queries/trends/lifecycle.py:30]: debug?

**Review by neilkakkar — APPROVED** [review]
one comment, looks good otherwise!

File: posthog/queries/trends/lifecycle.py (hunk 2 above)
Comment (psykomal) [reply posthog/queries/trends/lifecycle.py]: my bad, removed.

**Comment by psykomal** [comment]
@neilkakkar also should I run ./bin/tests to update snapshots for the failing tests or is there another way?

**Comment by neilkakkar** [comment]
That's the correct way. (`./bin/tests posthog/models/filters/test/test_filter.py::TestDjangoPropertiesToQ::test_array_property_as_string_on_persons` ) Just merge master though & it should go away 👀

**Comment by psykomal** [comment]
@neilkakkar updated snapshots. Please have a look. Not sure if I'm missing something but just merging didn't change anything.

File: posthog/api/test/dashboards/__snapshots__/test_dashboard.ambr [hunk 3]
```diff
@@ -2575,6 +2615,7 @@
          "posthog_team"."ingested_event",
          "posthog_team"."autocapture_opt_out",
          "posthog_team"."autocapture_exceptions_opt_in",
+         "posthog_team"."autocapture_exceptions_errors_to_ignore",
```
Comment (neilkakkar) [inline posthog/api/test/dashboards/__snapshots__/test_dashboard.ambr:2618]: cc: @pauldambra was this left out somehow by the job in the last PR?

Comment (pauldambra) [reply posthog/api/test/dashboards/__snapshots__/test_dashboard.ambr:2618]: Oh, I've seen this recently with some dashboard changes where snapshots in master seemed to be "out of sync" (for want of a better word) with PRs

https://github.com/PostHog/posthog/pull/16143

I didn't make time to track down the issue

**Comment by neilkakkar** [comment]
Seems like I can't merge this, will merge the duplicate I have here: https://github.com/PostHog/posthog/pull/16248 - it still references your commit though so should get attributed properly :)

**Comment by neilkakkar** [comment]
Thanks again for contributing! :D

**Pull request closed without merging**